content/blog/2023-07-31-going-out-on-a-limb-efficient-ecc-p384.markdown

```diff
@@ -0,0 +1,214 @@
+
+How is this possible? Well first let's quickly explain some language:
+
+## Elliptic Curves
+
+When it comes to cryptography, it's more than likely someone with a computer science background will at least be familiar with ideas such as key-exchange and private-key signing. The stand-in asymetric cipher in a typical computer science curriculum will almost certainly by RSA. However the hayday of Elliptic Curve ciphers has well and truly arrived, and their operation seems no less mystical than when they were just a toy for academia.
+
+I conjecture that programmers find ECC unnerving because the word 'Elliptic' implies some amount of continuous maths. Perhaps you've seen images of what an elliptic curve looks like as a subset of $\mathbb{R}^2$. As a useful cryptoraphic problem, we fundamentally are just interested with the algebraic properties of $\mathbb{F}$-rational points on the curve - an overly complex way to say 'the points with coordinates belonging to the [field](https://en.wikipedia.org/wiki/Field_(mathematics)) $\mathbb{F}$ that sit on the curve'. We can demonstrate their algebraic properties by an application of [Bézout's Theorem](https://en.wikipedia.org/wiki/Bézout%27s_theorem#:~:text=Bézout%27s%20theorem%20is%20a%20statement,the%20degrees%20of%20the%20polynomials.). What is not commonly understood is that the fields in use in ECC are very familiar mathematical objects.
+
+For example $\mathbb{F}_p$, that is the [unique](https://en.wikipedia.org/wiki/Finite_field#Existence_and_uniqueness) field with $p$ elements. The most straight-forward construction of this field is just to imagine arithmetic modulo $p$. The other finite fields used in practise in ECC are what some might call 'binary fields'. They're of the form $\mathbb{F}_{2^m}$, and will be familiar to you if you've studied AES (byte substitution is implemented by inversion modulo $\mathbb{F}_{2^8}$).
+
+If we allow ourselves to be reductive for just a moment, you might observe that all ECC math for a particular curve therefore can be boiled down to just efficiently implementing fixed-point arithmetic modulo a single prime constant, $p$. From here on out, I'll be speaking from this abstraction layer alone. 101 level discrete maths should be the only prereq (hopefully).
+
+## Limbs
+
+You wish to multiply two $m$ bit numbers, have subdivided such a number into $n$ machine words in some way. Let's suppose just for now that $n$ divides $m$ neatly, then the quotient $d$ is the minimum number of bits in each machine word that will be required for representing our number. Suppose we use the straight-forward representation whereby the least significant $d$ bits are used for storing parts of our number, which we better call $x$ because this is crypto and descriptive variable names are deemed harmful apparently.
```

Comment:
>numbers, have

should this be "numbers and have"?  I wouldn't expect the comma unless there was another comma later in the sentence

>are deemed harmful apparently

I would use "are apparently deemed harmful" or "are deemed harmful (apparently)"

Reply:
Trying to quote Edger Dijkstra so I'll use the latter :P